sallyport/src/lib.rs

```diff
@@ -179,24 +179,26 @@ impl Block {
     }
 
     /// Returns a Cursor for the Block
-    pub fn cursor(&mut self) -> Cursor<'_> {
-        Cursor(UnsafeCell::new(&mut self.buf))
+    pub fn cursor(&mut self) -> Cursor {
+        Cursor(&mut self.buf)
     }
 }
 
 /// Helper for allocation of untrusted memory in a Block.
-pub struct Cursor<'a>(UnsafeCell<&'a mut [u8]>);
+pub struct Cursor<'a>(&'a mut [u8]);
 
-impl Cursor<'_> {
+impl<'a> Cursor<'a> {
```

Comment:
```suggestion
impl Cursor<'_> {
```

Reply:
Because of the comment below, an explicit lifetime is required here.